File: Assuit-Sheet/Third-Sheet/X.py
def is_valid_cell(x, y, n, m):
    return x >= 0 and x < n and y >= 0 and y < m

def check_neighbors(matrix, x, y):
    n = len(matrix)
    m = len(matrix[0])
    neighbors = [(x-1, y-1), (x-1, y), (x-1, y+1), (x, y-1), (x, y+1), (x+1, y-1), (x+1, y), (x+1, y+1)]
    
    for nx, ny in neighbors:
        if is_valid_cell(nx, ny, n, m) and matrix[nx][ny] != 'x':
            return False
    return True

# Read input
n, m = map(int, input().split())
matrix = [list(input().strip()) for _ in range(n)]
x, y = map(int, input().split())

# Adjusting for 0-based indexing
x -= 1
y -= 1

# Check if all neighbors of the cell (x, y) are 'x'
if check_neighbors(matrix, x, y):
    print("yes")
else:
    print("no")

File: Assuit-Sheet/Third-Sheet/X.cpp
#include <iostream>
using namespace std;


int main()
{
    int N, M;
    cin >> N >> M;

    char A[N][M];

    for (int i = 0; i < N; i++)
    {
        for (int j = 0; j < M; j++)
        {
            cin >> A[i][j];
        }
    }

    int X, Y;
    cin >> X >> Y;
    X--;
    Y--;
    if (X >= 0 && X < N && Y >= 0 && Y < M)
    {
        if (( X - 1 == 0 || Y - 1 == 0) || (X - 1 ==0  || Y == 0) || (X - 1 ==0 || Y + 1 ==0) || (X ==0 || Y - 1 ==0) || (X ==0|| Y + 1 ==0) || (X + 1 ==0|| Y - 1 ==0) || (X + 1 ==0|| Y ==0) || (X + 1 ==0 || Y + 1 ==0))
        {
            if (A[X - 1][Y - 1] == 'x' && A[X - 1][Y] == 'x' && A[X - 1][Y + 1] == 'x' && A[X][Y - 1] == 'x' && A[X][Y + 1] == 'x' && A[X + 1][Y - 1] == 'x' && A[X + 1][Y] == 'x' && A[X + 1][Y + 1] == 'x')
            {
                cout << "YES";
            }
            else cout << "NO";
        }
        /* else if (A[X - 1][Y - 1] == 'x' && A[X - 1][Y] == 'x' && A[X - 1][Y + 1] == 'x' && A[X][Y - 1] == 'x' && A[X][Y + 1] == 'x' && A[X + 1][Y - 1] == 'x' && A[X + 1][Y] == 'x' && A[X + 1][Y + 1] == 'x')
        {
            cout << "YES";
        } */
            
    }

    return 0;
}
